gef.py

```diff
@@ -6523,9 +6523,20 @@ def find_tcache() -> int:
 
     @staticmethod
     def check_thread_ids(tids: List[int]) -> List[int]:
-        """Return the subset of tids that are currently valid."""
-        existing_tids = set(t.num for t in gdb.selected_inferior().threads())
-        return list(set(tids) & existing_tids)
+        """Check the validity, dedup, and return all valid tids."""
+        existing_tids = [t.num for t in gdb.selected_inferior().threads()]
+        valid_tids = set()
+        for tid in tids:
+            try:
+                tid = int(tid)
+            except ValueError:
+                err(f"Invalid thread id {tid:d}")
```

Comment:
`:d` would presumably fail since `int(tid)` failed. Why do we even have to do this?

Reply:
I dont know, I am just reverting to old logic xD

Reply:
Please don't just blindly revert.